src/DssSpell.sol

```diff
@@ -22,130 +22,392 @@ import "dss-exec-lib/DssAction.sol";
-        DssExecLib.sendPaymentFromSurplusBuffer(ORA_001_GAS_EMERGENCY, 1_805_407);
-
-
-        // Core Unit DAI Budget Streams
-        // https://mips.makerdao.com/mips/details/MIP40c3SP52
-        DssVestLike(MCD_VEST_DAI).restrict(
-            DssVestLike(MCD_VEST_DAI).create(DUX_001_WALLET,   1_934_300 * WAD, FEB_01_2022, JAN_31_2023 - FEB_01_2022,            0, address(0))
-        );
-        // https://mips.makerdao.com/mips/details/MIP40c3SP55
-        DssVestLike(MCD_VEST_DAI).restrict(
-            DssVestLike(MCD_VEST_DAI).create(SES_001_WALLET,   5_844_444 * WAD, FEB_01_2022, JAN_31_2023 - FEB_01_2022,            0, address(0))
-        );
-        // https://mips.makerdao.com/mips/details/MIP40c3SP47
-        DssVestLike(MCD_VEST_DAI).restrict(
-            DssVestLike(MCD_VEST_DAI).create(SNE_001_WALLET,     257_500 * WAD, FEB_01_2022, JUL_31_2023 - FEB_01_2022,            0, address(0))
-        );
-        // https://mips.makerdao.com/mips/details/MIP40c3SP53
-        DssVestLike(MCD_VEST_DAI).restrict(
-            DssVestLike(MCD_VEST_DAI).create(TECH_001_WALLET,  2_486_400 * WAD, FEB_01_2022, JAN_31_2023 - FEB_01_2022,            0, address(0))
-        );
-        // https://mips.makerdao.com/mips/details/MIP40c3SP46
-        DssVestLike(MCD_VEST_DAI).restrict(
-            DssVestLike(MCD_VEST_DAI).create(SF_001_WALLET,      494_502 * WAD, FEB_01_2022, JUL_31_2023 - FEB_01_2022,            0, address(0))
-        );
-        // https://forum.makerdao.com/t/rwf-001-auditor-flow/12900
-        DssVestLike(MCD_VEST_DAI).yank(15);
-        DssVestLike(MCD_VEST_DAI).restrict(
-            DssVestLike(MCD_VEST_DAI).create(RWF_001_WALLET,   1_705_000 * WAD, FEB_01_2022, DEC_31_2022 - FEB_01_2022,            0, address(0))
-        );
-
-
-        // Core Unit MKR Vesting Streams (sourced from treasury)
-        // https://mips.makerdao.com/mips/details/MIP40c3SP48
-        DssVestLike(MCD_VEST_MKR_TREASURY).restrict(
-            DssVestLike(MCD_VEST_MKR_TREASURY).create(
-                SF_001_VEST_01,  // Participant
-                240 * WAD,       // Amount
-                SEP_01_2021,     // Begin date
-                3 * 365 days,    // Vest duration
-                365 days,        // Cliff time
-                SF_001_WALLET    // Manager
-            )
-        );
-        // https://mips.makerdao.com/mips/details/MIP40c3SP48
-        DssVestLike(MCD_VEST_MKR_TREASURY).restrict(
-            DssVestLike(MCD_VEST_MKR_TREASURY).create(
-                SF_001_VEST_02,
-                240 * WAD,
-                APR_01_2021,
-                3 * 365 days,
-                365 days,
-                SF_001_WALLET
-            )
-        );
-
-
-        // Housekeeping
-        // Add CLIP_FAB to the Chainlog
-        DssExecLib.setChangelogAddress("CLIP_FAB", 0x0716F25fBaAae9b63803917b6125c10c313dF663);
+        address OLD_MCD_ESM = DssExecLib.getChangelogAddress(MCD_ESM);
+        address addr;
```

Comment:
Maybe instead of using the reusable `addr` is just better to load all the variables as immutable in the constructor of the action.

Reply:
as discussed will leave as is (for parity with the goerli casted spell among other reasons).